src/ttcontrol/ttcontrol.py

```diff
@@ -2,16 +2,31 @@
 # Copyright (C) 2024, Tiny Tapeout LTD
 
 import sys
+import machine
+import math
 
 print(f"\nversion={sys.version.split(';')[1].strip()}")
 
 def select_design(design):
    print(f"TODO: SELECT A DESIGN")
    print(f"design={design}")
 
-def set_clock_hz(hz):
-    print(f"TODO: SET CLOCK HZ")
-    print(f"hz={hz}")
+def set_clock_hz(hz, max_rp2040_freq=133_000_000):
+    # Only support integer frequencies
+    freq = int(hz)
+
+    print(f"Setting clock frequency to {freq} Hz")
+
+    # Get best acheivable RP2040 clock rate for that rate
+    rp2040_freq = _get_best_rp2040_freq(freq, max_rp2040_freq)
+
+    # Apply the settings
+    machine.freq(rp2040_freq)
+
+    # Clock pin is pin 0
+    pwm = machine.PWM(0)
```

Comment:
Let's put the clock pin number into a constant (then we can also remove the comment above)